src/logic/tokens/store/actions/fetchTokens.js

```diff
@@ -37,25 +39,119 @@ const createERC721TokenContract = async () => {
   return erc721Token
 }
 
+const OnlyBalanceToken = {
+  contractName: 'OnlyBalanceToken',
+  abi: [
+    {
+      constant: true,
+      inputs: [
+        {
+          name: 'owner',
+          type: 'address',
+        },
+      ],
+      name: 'balanceOf',
+      outputs: [
+        {
+          name: '',
+          type: 'uint256',
+        },
+      ],
+      payable: false,
+      stateMutability: 'view',
+      type: 'function',
+    },
+    {
+      constant: true,
+      inputs: [
+        {
+          name: 'owner',
+          type: 'address',
+        },
+      ],
+      name: 'balances',
+      outputs: [
+        {
+          name: '',
+          type: 'uint256',
+        },
+      ],
+      payable: false,
+      stateMutability: 'view',
+      type: 'function',
+    },
+  ],
+}
+
+// For the `batchRequest` of balances, we're just using the `balanceOf` method call.
+// So having a simple ABI only with `balanceOf` prevents errors
+// when instantiating non-standard ERC-20 Tokens.
+const createOnlyBalanceToken = () => {
+  const web3 = getWeb3()
+  const contract = new web3.eth.Contract(OnlyBalanceToken.abi)
+  return contract
+}
+
 export const getHumanFriendlyToken = ensureOnce(createHumanFriendlyTokenContract)
 
 export const getStandardTokenContract = ensureOnce(createStandardTokenContract)
 
 export const getERC721TokenContract = ensureOnce(createERC721TokenContract)
 
+export const getOnlyBalanceToken = ensureOnce(createOnlyBalanceToken)
+
 export const containsMethodByHash = async (contractAddress: string, methodHash: string) => {
   const web3 = getWeb3()
   const byteCode = await web3.eth.getCode(contractAddress)
 
   return byteCode.indexOf(methodHash.replace('0x', '')) !== -1
 }
 
-export const fetchTokens = () => async (dispatch: ReduxDispatch<GlobalState>) => {
+export const getTokenInfos = async (tokenAddress: string) => {
+  if (!tokenAddress) {
+    return null
+  }
+  const { tokens } = store.getState()
+  const localToken = tokens.get(tokenAddress)
+  // If the token is inside the store we return the store token
+  if (localToken) {
+    return localToken
+  }
+  // Otherwise we fetch it, save it to the store and return it
+  const tokenContract = await getHumanFriendlyToken()
+  let tokenInstance = await tokenContract.at(tokenAddress)
```

Comment:
this should be `const`, it's not reassigned anymore

Reply:
done!